### DISCUSSION TITLE
Kubernetes controller to automatically order certs for new/changed ingresses

### DISCUSSION THREAD

**Discussion by andreasthuen** [polymind-inc/acmebot#726] [General]
Hi everyone! We have been using this project for about a year now and it works very well for us :) 

I was not sure what the best place to post this would be, I hope that here is OK 😄 
We have created and open sourced a simple kubernetes controller that runs as a pod inside a cluster and watches any new or changed ingress objects.

Please check it out and let me know if you have any questions or comments 💬 
[ingress2acmebotreflector](https://github.com/SparebankenVest/ingress2acmebotreflector)

This is how it works:

1. Whenever it detects a new or changed ingress it calls the acmebot API and checks if a valid certificate exists for the given DNS.
2. If it does not exists, a new cert is ordered using the acmebot API.
3. The cert is then imported to the cluster from the keyvault using [akv2k8s](https://akv2k8s.io/)

**Comment by JoeBawn** [comment]
Hi @andreasthuen!

We have recently started using keyvault-acmebot and are already using akv2k8s on both AKS and on K8 environments. We currently have a process to issue certs using the api, however I have been looking into using the the ingress2acmebotrefelector you have created to further enhance the process, but have been having some issues with authentication and not being able to retrieve a token. 

Would be great to get it up and running so if you have any suggestions would be greatly appreciated. 

Thanks
Joe
